Lib/Schema/Table.php

```diff
@@ -0,0 +1,603 @@
+          return $this->buildCreate();
+          break;
+        case 'drop':
+          return $this->buildDrop();
+          break;
+        case 'alter':
+          return $this->buildAlter();
+          break;
+      }
+    } catch (Exception $e) {
+      throw $e;
+    }
+  }
+
+  /**
+   * Alter table.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @param array $fields Fields to be altered.
+   * @param string $type Alter type. Value can be 'add' or 'drop'.
+   *
+   * @return self
+   *
+   * @throws Freyja\Exceptions\InvalidArgumentException if some element in
+   * $fields isn't a Freyja\Database\Schema\Field object.
+   * @throws \LogicException if $fields is an empty array.
+   */
+  private function alter(array $fields, $type) {
+    if (empty($fields))
+      throw new LogicException('It\'s required at least one column to alter the table');
+    foreach ($fields as $field)
+      if (!is_a($field, 'Freyja\Database\Schema\Field'))
+        throw InvArgExcp::typeMismatch('field', $field, 'Freyja\Database\Schema\Field');
+
+    $this->type = 'alter';
+    $this->alter_fields[$type] = array_merge($this->alter_fields, $fields);
+    return $this;
+  }
+
+  /**
+   * Build 'create' query string.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @return string
+   *
+   * @throws \RuntimeException if it is raised by the Field method called.
+   * @see Freyja\Database\Schema\Field::getField()
+   * @throws \LogicException if AUTO_INCREMENT is set on more than one field, or
+   * in a non primary key field, or if $fields property is an empty array.
+   */
+  private function buildCreate() {
+    $query = 'CREATE TABLE IF NOT EXISTS '.$this->name.' (';
+
+    $count = 0;
+    $autoinc_fields = 0;
+    if (empty($fields))
+      throw new LogicException('A table must have at least 1 column');
+    foreach ($this->fields as $field) {
+      if ($count != 0)
+        $query .= ', ';
+
+      try {
+        $field_info = $field->getField();
+      } catch (Exception $e) {
+        throw $e;
+      }
+      $field_name = key($field_info);
+      $info = $field_info[$field_name];
+
+      // Append field name and type.
+      $query .= $field_name.' '.$info['type'];
+
+      // Append default value, if set.
+      if (!is_null($info['default']))
+        $query .= ' DEFAULT '.$info['default'];
+
+      // Append NOT NULL, if true.
+      if ($info['NOT NULL'])
+        $query .= ' NOT NULL';
+
+      // Append UNSIGNED, if true.
+      if ($info['UNSIGNED'])
+        $query .= ' UNSIGNED';
+
+      // Append AUTO_INCREMENT, if true.
+      if ($info['AUTO_INCREMENT']) {
+        if ($autoinc_fields > 0)
+          throw new LogicException('AUTO_INCREMENT cannot be set on more than one field');
+        if (!in_array($info['name'], $this->primary_keys))
+          throw new LogicException('AUTO_INCREMENT cannot be set on a field that isn\'t primary key');
+
+        $query .= ' AUTO_INCREMENT';
+      }
+    }
```

Comment:
Cannot do it, I need to create every single field in the table, so i can manage the `AUTO_INCREMENT` problem. I can't know how many of them there are, and if they are primary key or not, from inside a single field.